text/0000-build-std/4-stage-1a.md

```diff
@@ -0,0 +1,1061 @@
+# Proposal
+[proposal]: #proposal
+
+Cargo configuration will contain a new key `build-std` under the `[build]`
+section ([?][rationale-build-std-in-config]), permitting one of two values -
+"never" ([?][rationale-build-std-never]) or "always", defaulting to "never":
+
+```toml
+[build]
+build-std = "always" # or `never`
+```
+
+`build-std` can also be specified in the `[target.<triple>]` and
+`[target.<cfg>]` sections ([?][rationale-build-std-target-section]):
+
+```toml
+[target.aarch64-unknown-illumos]
+build-std = "always" # or `never`
+```
+
+The `build-std` configuration locations have the following precedence
+([?][rationale-build-std-precedence]):
+
+1. `[target.<triple>]`
+2. `[target.<cfg>]`
+3. `[build]`
+
+As the Cargo configuration is local to the current installation of Cargo
+(typically in `~/.config/cargo`), the value of `build-std` is not influenced by
+the dependencies of the current crate.
+
+When `build-std` is set to "always", then the standard library will be
+unconditionally recompiled ([?][rationale-unconditional]) in the release profile
+defined in its workspace as part of every clean build
+([?][rationale-release-profile]). This is primarily useful for users of tier
+three targets. As with other dependencies, the standard library's build will
+respect the `RUSTFLAGS` environment variable.
+
+> [!NOTE]
+>
+> Configuration of the pre-built standard library is split across bootstrap and
+> the Cargo packages for the standard library. As much of this configuration as
+> possible should be moved to the Cargo profile for these packages so that the
+> artifacts produced by build-std match the pre-built standard library as much
+> as is feasible.
+
+Alongside `build-std`, a `build-std-crate` key will be introduced
+([?][rationale-build-std-crate]), which can be used to specify which crates from
+the standard library is to be built. Only "core", "alloc" and "std" are valid
+values for `build-std-crate`.
+
+```toml
+[build]
+build-std-crate = "std"
+```
+
+If [*Stage 1b* of this proposal][stage1b] is implemented then `build-std-crate`
+will not be used unless explicitly set and the crate graph's dependencies on the
+standard library will determine which crates are built instead. Otherwise,
+`build-std-crate` will default to "std".
+
+If `std` is to be built and Cargo is building a test using the default test
+harness then Cargo will also build the `test` crate.
```

Comment:
What about non-default test harnesses, like https://crates.io/crates/defmt-test?

Reply:
I think these would work with the RFC as written - when the default test harness is disabled with `harness = false`, we won't build it, and then you'll still have the non-default harness as a `dev-dependency`?